examples/request_state_local_cache/src/main.rs

```diff
@@ -0,0 +1,62 @@
+#![feature(plugin, decl_macro)]
+#![plugin(rocket_codegen)]
+extern crate rocket;
+
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use rocket::request::{self, Request, FromRequest, State};
+use rocket::outcome::Outcome::*;
+
+#[cfg(test)] mod tests;
+
+struct Atomics {
+    first: AtomicUsize,
+    second: AtomicUsize,
+}
+
+struct Guard1();
+struct Guard2();
+
+impl<'a, 'r> FromRequest<'a, 'r> for Guard1 {
+    type Error = ();
+
+    fn from_request(req: &'a Request<'r>) -> request::Outcome<Self, ()> {
+        req.guard::<State<Atomics>>()?.first.fetch_add(1, Ordering::Relaxed);
+        req.local_cache(|req| {
```

Comment:
This is saving a value of type `()` in the cache. I suppose that's fine, but let's at least leave a comment indicating this.